CentroidCalculatorTests.testAddDifferentDimensionalType failure

This failed in a PR build, but fails on master reproducibly with the given seed.

**Build scan**:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-2/14709/testReport/junit/org.elasticsearch.xpack.spatial.index.fielddata/CentroidCalculatorTests/testAddDifferentDimensionalType/

**Repro line**:
```
./gradlew ':x-pack:plugin:spatial:test' --tests "org.elasticsearch.xpack.spatial.index.fielddata.CentroidCalculatorTests.testAddDifferentDimensionalType" -Dtests.seed=CE8C973B645ED105 -Dtests.security.manager=true -Dtests.locale=lv-LV -Dtests.timezone=America/Kentucky/Louisville -Druntime.java=11
```

**Reproduces locally?**:
Yes

**Applicable branches**:
master

**Failure excerpt**:
```
java.lang.AssertionError: 
Expected: <0.0>
     but: was <-20.333333333333332>
	at __randomizedtesting.SeedInfo.seed([CE8C973B645ED105:39316408FD9D4718]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.xpack.spatial.index.fielddata.CentroidCalculatorTests.testAddDifferentDimensionalType(CentroidCalculatorTests.java:359)
```

--- a/x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/fielddata/CentroidCalculatorTests.java
+++ b/x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/fielddata/CentroidCalculatorTests.java
@@ -356,9 +356,11 @@ public class CentroidCalculatorTests extends ESTestCase {
             CentroidCalculator calculator = new CentroidCalculator();
             calculator.add(point);
             calculator.add(line);
-            assertThat(calculator.getX(), equalTo(lineCalculator.getX()));
-            assertThat(calculator.getY(), equalTo(lineCalculator.getY()));
-            assertThat(calculator.sumWeight(), equalTo(lineCalculator.sumWeight()));
+            if (lineCalculator.getDimensionalShapeType() == LINE) {  // skip degenerated line
+                assertThat(calculator.getX(), equalTo(lineCalculator.getX()));
+                assertThat(calculator.getY(), equalTo(lineCalculator.getY()));
+                assertThat(calculator.sumWeight(), equalTo(lineCalculator.sumWeight()));
+            }
         }
 
         // line add point
@@ -368,9 +370,11 @@ public class CentroidCalculatorTests extends ESTestCase {
             CentroidCalculator calculator = new CentroidCalculator();
             calculator.add(line);
             calculator.add(point);
-            assertThat(calculator.getX(), equalTo(lineCalculator.getX()));
-            assertThat(calculator.getY(), equalTo(lineCalculator.getY()));
-            assertThat(calculator.sumWeight(), equalTo(lineCalculator.sumWeight()));
+            if (lineCalculator.getDimensionalShapeType() == LINE) { // skip degenerated line
+                assertThat(calculator.getX(), equalTo(lineCalculator.getX()));
+                assertThat(calculator.getY(), equalTo(lineCalculator.getY()));
+                assertThat(calculator.sumWeight(), equalTo(lineCalculator.sumWeight()));
+            }
         }
 
         // line add line

